tools/src/main/java/org/bitcoinj/tools/BuildCheckpoints.java

```diff
@@ -90,31 +98,44 @@ public static void main(String[] args) throws Exception {
                 throw new RuntimeException("Unreachable.");
         }
 
+
+        // Configure bitcoinj to fetch only headers, not save them to disk, connect to a local fully synced/validated
+        // node and to save block headers that are on interval boundaries, as long as they are <1 month old.
+        final BlockStore store = new MemoryBlockStore(params);
+        final BlockChain chain = new BlockChain(params, store);
+        final PeerGroup peerGroup = new PeerGroup(params, chain);
+
         final InetAddress ipAddress;
         if (options.has("peer")) {
             String peerFlag = (String) options.valueOf("peer");
             try {
                 ipAddress = InetAddress.getByName(peerFlag);
+                startPeerGroup(peerGroup, ipAddress);
             } catch (UnknownHostException e) {
                 System.err.println("Could not understand peer domain name/IP address: " + peerFlag + ": " + e.getMessage());
                 System.exit(1);
                 return;
             }
+        } else if (dnsNetworks.contains(params.getClass())) {
```

Comment:
If you want to know if a given network has DNS seeds, why don't you use params.getDnsSeeds(). It will return null if it doesn't support DNS.